lib/AST/Expr.cpp

```diff
@@ -1770,23 +1770,20 @@ Expr *CallExpr::getDirectCallee() const {
   }
 }
 
-void ExplicitCastExpr::setCastType(Type type) {
-  CastTy->setType(MetatypeType::get(type));
-}
-
 ForcedCheckedCastExpr *ForcedCheckedCastExpr::create(ASTContext &ctx,
                                                      SourceLoc asLoc,
                                                      SourceLoc exclaimLoc,
                                                      TypeRepr *tyRepr) {
-  return new (ctx) ForcedCheckedCastExpr(nullptr, asLoc, exclaimLoc,
-                                         new (ctx) TypeExpr(tyRepr));
+  return new (ctx) ForcedCheckedCastExpr(nullptr, asLoc, exclaimLoc, tyRepr);
 }
 
 ForcedCheckedCastExpr *
 ForcedCheckedCastExpr::createImplicit(ASTContext &ctx, Expr *sub, Type castTy) {
+  assert(!castTy.isNull() && "Implicit cast without a type?");
   auto *const expr = new (ctx) ForcedCheckedCastExpr(
-      sub, SourceLoc(), SourceLoc(), TypeExpr::createImplicit(castTy, ctx));
+      sub, SourceLoc(), SourceLoc(), nullptr);
```

Comment:
Nit: could you please add `/*repr=*/` in places where there is `nullptr` used for the argument? Doesn't have to be in this PR, could be in a follow-up to.